src/FindAnApprenticeship/SFA.DAS.FindAnApprenticeship.Api/Controllers/UsersController.cs

```diff
@@ -85,5 +88,51 @@ public async Task<IActionResult> PostcodeAddress([FromQuery] string postcode)
                 return new StatusCodeResult((int)HttpStatusCode.InternalServerError);
             }
         }
+
+        [HttpGet]
+        [Route("select-address")]
+        public async Task<IActionResult> SelectAddress([FromQuery] string postcode)
+        {
+            try
+            {
+                var queryResponse = await _mediator.Send(new GetCandidateAddressesByPostcodeQuery(postcode));
+
+                if (queryResponse.AddressesResponse == null || !queryResponse.AddressesResponse.Addresses.Any())
+                    return NotFound();

```

Comment:
We should really just return an empty list in this case rather than a not found result.